### DISCUSSION TITLE
Is there a way to use storybook to document helper functions/React Hooks in your react library?

### DISCUSSION THREAD

**Discussion by chadlavi-casebook** [storybookjs/storybook#15512] [Help]
For example, I have several helper functions and React Hooks that are used throughout my TypeScript React lib. They don't return jsx, they return strings, booleans, sometimes objects. You know, helper function stuff. Point being, they're not components, so I can't document them the way I would document a react component.

How can I incorporate these into my storybook documentation without just manually writing documentation for them that would need to be updated manually if they are ever edited? I have JSDoc comments on these functions, so I know that in theory react docgen could pick them up. Just don't know how to get that to work in Storybook.

**Comment by goszczynskip** [comment]
I'm putting hooks in `ArgsTable` as subcomponents. Just remember to have a `.tsx` extension instead of `.ts` on your hook file. I couldn't make it work with the `.ts` extension.

**Reply by RichardHpa** [reply]
This would be a great thing for `ArgsTable` to expand on. I ideally want to use it for our custom hooks. Currently I have a seperate file like @goszczynskip recommends but it seems really annoying to do that.

**Reply by ccsenter** [reply]
I realize this was a long time ago, but I'm also facing the same issue: how to document the return type of a utility function or hook?
@goszczynskip can you elaborate on what your solution/workaround was? (_"putting hooks in ArgsTable as subcomponents_")
